### DISCUSSION TITLE
[HELP] How to Evaluate HE position in the Screen

### DISCUSSION THREAD

**Discussion by FelipeRearden** [nothingislost/obsidian-hover-editor#218] [Q&A] [answered]
Hello @pjeby 

As we know we can change the position of a HE in  the screen:
→ Snap to Right
→ Snap to Left
→ Minimize
→ Viewport (Maximize)
→ Restore

**My HELP REQUEST is related to this:**

Is it possible to evaluate the position that the active HE is in the screen?

### MY Idea 

I wanna create a Templater Script to create a command where I can **Cycle** between some views: `Snap to Right → Viewport → Minimize →  Snap to Right` with a single command.

I am thinking something like this (forgive for my bad code skills)

```js
if (HE.direction = "Right") {
app.commands.executeCommandById('obsidian-hover-editor:snap-active-popover-to-viewport')
} else if (HE.direction = "Viewport") {
app.commands.executeCommandById('obsidian-hover-editor:minimize-active-popover')
} else if (HE.direction = "Minimize") {
app.commands.executeCommandById('obsidian-hover-editor:snap-active-popover-to-right')
}
```

Is it feasible to create something like this? If Yes, would you mind helping me with the code to achieve this? 🙏🙏🙏

Thanks for reading this and have a great day.

**Comment by pjeby** [comment, marked as answer]
Something like this ought to do:

```js
const he = app.workspace.activeLeaf?.containerEl.matchParent('.hover-editor');
if (!he) {
    // active tab is not a hover editor
} else if (he.matches(".snap-to-right")) {
} else if (he.matches(".snap-to-left")) {
} else if (he.matches(".snap-to-viewport")) {
} else if (he.matches(".is-minimized")) {
} else {
    // hover editor, but not snapped or minimized
}
```

Note that it's possible to be both minimized and snapped, but in that case the snapping takes precedence and it will return to minimized state once unsnapped.

**Reply by FelipeRearden** [reply]
Thank you very much @pjeby !!!!

Working 100% perfect. This is exactly what I was looking for. 

Thank you very much for your help and I wish you a fantastic day!
